### PR TITLE
Create test framework for operators

### PR DESCRIPTION
**Pull request by Bronek**
This is opinionated test framework for kubernetes operators. It is based on https://github.com/dlespiau/kube-test-harness/  (thank you!) but with added
* protocol for starting up and shutting down the K8s environment.
* protocol for starting up and shutting down an operator being tested

Both protocols are (in this version) based on make targets, where the operator code being tested provides Makefile targets such as `tests-cluster-start` `tests-operator-start` etc. (the "tests-" prefix can be overridden in options).

I plan to make the operator start delay configurable (currently hardcoded to 2s). Also, this will eventually support tests written with https://github.com/DATA-DOG/godog , but vanilla `go test` will be always supported.

### CODE REVIEW

**Comment by Bronek** [comment]
Will investigate CI failures tomorrow, for now just please give your feedback, if you have any

File: pkg/test_framework/test.go [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+func (t *Test) StopOperator() {
+	if t.stopOperator {
+		stopOperator(t.harness.Options, t.harness.Sinks)
+		t.stopOperator = false
+	}
+}
+
+func (t *Test) DeleteDeployment(d *appsv1.Deployment, timeout time.Duration) {
+	t.Test.DeleteDeployment(d)
+	t.Test.WaitForDeploymentDeleted(d, timeout)
```
Comment (keymone) [inline pkg/test_framework/test.go]: probably useful to return timeout error, if timed out

File: pkg/test_framework/harness.go [hunk 2]
```diff
@@ -0,0 +1,252 @@
...
+	if err != nil {
+		log.Panic(err)
+		return
+	}
+	log.Print("... done")
+}
+
+func stopCluster(options Options, sinks Sinks) {
+	// Do not stop the cluster if panicking, to enable troubleshooting
+	if r := recover(); r != nil {
```
[the remark this answers is not in the document]
Comment (Bronek) [reply pkg/test_framework/harness.go]: We do not use panic/recover for control flow; we merely handle panics inside the tests gracefully, e.g. `t.Fail()` will do that.

File: pkg/test_framework/test.go (hunk 1 above)
Comment (Bronek) [reply pkg/test_framework/test.go]: I have to rely on kube-test-harness for such functionality - and it will fail test if timeout occurred.

**Pull request merged**